PathPlanning/BreadthFirstSearch/breadth_first_search.py

```diff
@@ -0,0 +1,265 @@
+                ngoal.cost = current.cost
+                break
+
+            # Remove the item from the open set
+            del open_set[c_id]
+
+            random.shuffle(self.motion)
+
+            # expand_grid search grid based on motion model
+            for i, _ in enumerate(self.motion):
+                node = self.Node(current.x + self.motion[i][0],
+                                 current.y + self.motion[i][1],
+                                 current.cost + self.motion[i][2], c_id+1, None)
+                n_id = self.calc_grid_index(node)
+
+                # If the node is not safe, do nothing
+                if not self.verify_node(node):
+                    continue
+
+                if n_id not in closed_set:
+                    node.parent = current
+                    closed_set[n_id] = node
+                    open_set[n_id] = node
+
+        rx, ry = self.calc_final_path(ngoal, closed_set)
+        return rx, ry
+
+    def calc_final_path(self, ngoal, closedset):
+        # generate final course
+        rx, ry = [self.calc_grid_position(ngoal.x, self.minx)], [
+            self.calc_grid_position(ngoal.y, self.miny)]
+        n = closedset[ngoal.pind]
+        while n is not None:
+            rx.append(self.calc_grid_position(n.x, self.minx))
+            ry.append(self.calc_grid_position(n.y, self.miny))
+            n = n.parent
+
+        return rx, ry
+
+    def calc_grid_position(self, index, minp):
+        """
+        calc grid position
+
+        :param index:
+        :param minp:
+        :return:
+        """
+        pos = index * self.reso + minp
+        return pos
+
+    def calc_xyindex(self, position, min_pos):
+        return round((position - min_pos) / self.reso)
+
+    def calc_grid_index(self, node):
+        return (node.y - self.miny) * self.xwidth + (node.x - self.minx)
+
+    def verify_node(self, node):
+        px = self.calc_grid_position(node.x, self.minx)
+        py = self.calc_grid_position(node.y, self.miny)
+
+        if px < self.minx:
+            return False
+        elif py < self.miny:
+            return False
+        elif px >= self.maxx:
+            return False
+        elif py >= self.maxy:
+            return False
+
+        # collision check
+        if self.obmap[node.x][node.y]:
+            return False
+
+        return True
+
+    def calc_obstacle_map(self, ox, oy):
+
+        self.minx = round(min(ox))
+        self.miny = round(min(oy))
+        self.maxx = round(max(ox))
+        self.maxy = round(max(oy))
+        print("minx:", self.minx)
+        print("miny:", self.miny)
+        print("maxx:", self.maxx)
+        print("maxy:", self.maxy)
+
+        self.xwidth = round((self.maxx - self.minx) / self.reso)
+        self.ywidth = round((self.maxy - self.miny) / self.reso)
+        print("xwidth:", self.xwidth)
+        print("ywidth:", self.ywidth)
+
+        # obstacle map generation
+        self.obmap = [[False for i in range(self.ywidth)]
```

Comment:
i -> _

Reply:
Just used the one already built in A*, so I am modifying it there as well ?

Reply:
Done

Reply:
Did you change?

Reply:
Oops, changed on the other branches but not this one.